### PR TITLE
dar: cleanup

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/dar/default.nix [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
 }:
 
-with lib;
-
 let
-  # Fails to build with clang on Darwin:
+  # Fails to build with clang-11 on Darwin:
   # error: exception specification of overriding function is more lax than base version
-  stdenv = gccStdenv;
+  stdenv = if args.stdenv.isDarwin then llvmPackages_12.stdenv else args.stdenv;
```
Comment (wegank) [inline pkgs/tools/backup/dar/default.nix:27]: We can move this to `all-packages.nix`, i.e.

```
  dar = callPackage ../tools/backup/dar {
    stdenv = if stdenv.isDarwin then llvmPackages_12.stdenv else stdenv;
  };
```

Comment (Atemu) [reply pkgs/tools/backup/dar/default.nix:27]: I know. I prefer to keep the logic in the packages' files because all-packages is a giant mess and I'd rather not add to that.  
I also think it's easy to forget that the `callPackage` site has any special logic in it because that's not visible from the files that are supposed to declare the derivation.

File: pkgs/tools/backup/dar/default.nix [hunk 2]
```diff
@@ -1,41 +1,62 @@
-{ lib, gccStdenv, fetchurl
-, which
-, attr, e2fsprogs
-, curl, libargon2, librsync, libthreadar
-, gpgme, libgcrypt, openssl
-, bzip2, lz4, lzo, xz, zlib
-, CoreFoundation
```
[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/tools/backup/dar/default.nix:7]: Order is preserved. Do you mean that I add some space between the "groups"?

[the remark this answers is not in the document]
Comment (Izorkin) [reply pkgs/tools/backup/dar/default.nix:7]: No, I wanted to leave something like this variant:
```
args @ { lib, gccStdenv, fetchzip, which
  , attr, e2fsprogs # packages used only in linux
  , curl, libargon2, librsync, libthreadar # libraries
  , gpgme, libgcrypt, openssl # encryption libraries
  , bzip2, lz4, lzo, xz, zlib # compression libraries
```
Can be left without comments.

[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/tools/backup/dar/default.nix:7]: Not really sure what exactly you mean.

I ported the multiple inputs per line to one input per line. This was on purpose since it makes files more readable and changes more composable.

File: pkgs/tools/backup/dar/default.nix [hunk 3]
```diff
@@ -47,20 +45,18 @@ stdenv.mkDerivation rec {
...
   ];
 
+  hardeningDisable = [ "format" ];
+
   postInstall = ''
     # Disable html help
     rm -r "$out"/share/dar
   '';
 
-  enableParallelBuilding = true;
```
Comment (Mindavi) [inline pkgs/tools/backup/dar/default.nix:55]: Why was this removed? I don't think it's enabled by default unless you use `cmake`, `meson`, `ninja` or something like it

**Review by Mindavi — APPROVED** [review]
One question, otherwise looks fine to me

File: pkgs/tools/backup/dar/default.nix (hunk 3 above)
Comment (Atemu) [reply pkgs/tools/backup/dar/default.nix:55]: Oh! I thought that option was always enabled by default and that's why I removed it. Nice catch.

**Review by Izorkin — APPROVED** [review]
